csharp/src/Apache.Arrow/Ipc/ArrowStreamReaderImplementation.cs

```diff
@@ -26,12 +27,14 @@ internal class ArrowStreamReaderImplementation : ArrowReaderImplementation
         public Stream BaseStream { get; }
         protected ArrayPool<byte> Buffers { get; }
         private readonly bool _leaveOpen;
+        private readonly MemoryPool _memoryPool;
 
-        public ArrowStreamReaderImplementation(Stream stream, bool leaveOpen)
+        public ArrowStreamReaderImplementation(Stream stream, MemoryPool memoryPool, bool leaveOpen)
         {
             BaseStream = stream;
+            _memoryPool = memoryPool ?? MemoryPool.Default.Value;
             _leaveOpen = leaveOpen;
-            Buffers = ArrayPool<byte>.Create();
+            Buffers = ArrayPool<byte>.Shared;
```

Comment:
Does it need to be a property?  It'd be more efficient to just have all callers use `ArrayPool<byte>.Shared.Rent/Return`, both because it would remove the extra field from this instance and because the .NET Core JIT will likely be able to devirtualize those calls.

Reply:
Fixed.